### ISSUE TITLE
Layout styles partially loaded when deploying to Netlify

### ISSUE THREAD

**Issue by bjfresh** [gatsbyjs/gatsby#12226]
## Description

I'm having an issue where Layout styles are only partially loading when deploying to Netlify, while CSS modules are loading fine. The deploy itself is completing without error. By partially loading, I mean I have a core SASS file with custom properties and breakpoints defined, where I'm also importing a small number of SASS files which apply styles to third-party components where I don't have control over HTML and classNames.

Breakpoints and custom properties seem to be applied on Netlify, but styling for the menu and swiper/slider components seem to be missing completely. Otherwise, I'm using CSS modules elsewhere without issue. Everything works when developing locally.

(The weird three white bars are the hamburger menu being stretched across the whole viewport :/ )

I've cleared the cache / deleted + re-installed node_modules, to no effect.

Thanks in advance!

### Steps to reproduce

The core SASS file is all.sass:
```css
\:root  // SASS requires backslash before ':'
  // Grid
  --max-width: 1248px
  --gutter-width: 18px

  // Colors
  --brand-brown: #3E2A0A
  --brand-yellow: #E4B114
  --brand-turquoise: #52D1DC
  --brand-green: #26532B
  --brand-blue: #124E78
  --brand-red: #C41E3D

  // Spacing
  --spacing-xs: 2px
  --spacing-sm: 4px
  --spacing-md: 8px
  --spacing-lg: 16px
  --spacing-xl: 32px
  --spacing-xxl: 64px
  --spacing-xxxl: 128px

@import 'breakpoints'
@import 'fonts'
@import 'menu'
@import 'swiper'
@import 'slider'

html
  font-family: 'Lato', sans-serif

body
  margin: 0
  -webkit-font-smoothing: antialiased
  -moz-osx-font-smoothing: grayscale
  background-color: var(--brand-yellow)
  transition: all 0.2s ease-in-out
  &.bistro
    background-color: var(--brand-red)
  &.lodge
    background-color: var(--brand-blue)
  &.groups-and-packages
    background-color: var(--brand-brown)
  &.yoho-national-park
    background-color: var(--brand-green)

figure
  margin: 0

.center
  margin-left: auto
  margin-right: auto
```

See https://trufflepigs.netlify.com/

### Expected result

All styles are loaded

### Actual result

Styles are not applied to menu or slider components (both of which are implemented by node modules)

### Environment

  System:
    OS: macOS 10.14.4
    CPU: (4) x64 Intel(R) Core(TM) i5-6287U CPU @ 3.10GHz
    Shell: 3.2.57 - /bin/bash
  Binaries:
    Node: 9.5.0 - ~/.nvm/versions/node/v9.5.0/bin/node
    Yarn: 1.12.3 - /usr/local/bin/yarn
    npm: 6.7.0 - ~/.nvm/versions/node/v9.5.0/bin/npm
  Languages:
    Python: 2.7.15 - /usr/local/bin/python
  Browsers:
    Chrome: 72.0.3626.119
    Firefox: 64.0.2
    Safari: 12.1
  npmPackages:
    gatsby: ^2.0.0 => 2.1.19 
    gatsby-awesome-pagination: ^0.3.3 => 0.3.3 
    gatsby-image: ^2.0.26 => 2.0.30 
    gatsby-plugin-lodash: ^3.0.1 => 3.0.4 
    gatsby-plugin-netlify: ^2.0.0 => 2.0.11 
    gatsby-plugin-purgecss: ^2.4.0 => 2.4.0 
    gatsby-plugin-react-helmet: ^3.0.0 => 3.0.7 
    gatsby-plugin-sass: ^2.0.1 => 2.0.10 
    gatsby-plugin-sharp: ^2.0.15 => 2.0.23 
    gatsby-plugin-svg-sprite: ^2.0.1 => 2.0.1 
    gatsby-remark-images: ^2.0.1 => 2.0.6 
    gatsby-source-filesystem: ^2.0.1 => 2.0.23 
    gatsby-source-wordpress: ^3.0.1 => 3.0.41 
    gatsby-transformer-remark: ^2.1.1 => 2.3.0 
    gatsby-transformer-sharp: ^2.1.1 => 2.1.15 
  npmGlobalPackages:
    gatsby-cli: 2.4.12

**Comment by DSchau** [comment]
Hi @bjfresh!

Sorry to hear you're running into an issue. To help us best begin debugging the underlying cause, it is incredibly helpful if you're able to create a [minimal reproduction][reproduction]. This is a simplified example of the issue that makes it clear and obvious what the issue is and how we can begin to debug it.

If you're up for it, we'd very much appreciate if you could [provide a minimal reproduction][reproduction] and we'll be able to take another look.

Thanks for using Gatsby! 💜

[reproduction]: https://gatsby.dev/reproduction

**Comment by bjfresh** [comment]
The site itself is fairly minimal, with the issue seemingly in the core 

I've made the repository public for now at https://github.com/bjfresh/trufflepigs.com

It's based on [Gatsby Starter WordPress ](https://www.gatsbyjs.org/starters/GatsbyCentral/gatsby-starter-wordpress/)

Almost all of the styles are in CSS modules, and each of these are working fine.
The issue is in a small number of imports in **all.css**:

```css
@import 'menu'
@import 'swiper'
@import 'slider'
```

Otherwise, the CSS custom properties and directives are being followed:
- Grid width and gutter
- Theme colours
- Spacing variables
- Breakpoints
- Fonts

Further info:
The menu is built with **react-burger-menu**
The slider is built with [React ID Swiper](http://kidjp85.github.io/react-id-swiper/)

**Comment by anantoghosh** [comment]
This looks like an issue caused by `gatsby-plugin-purgecss` .
First, update the plugin to the latest version.
Then try these options to configure `gatsby-plugin-purgecss`
```js
{
      resolve:'gatsby-plugin-purgecss', // purges all unused/unreferenced css rules
      options: {
        develop: true,            // Activates purging in npm run develop
        purgeOnly: ['/all.sass'], // applies purging only on these files
      },
    },
```

Furthermore read the documentation https://github.com/anantoghosh/gatsby-plugin-purgecss if you continue to face issues.

This issue does not belong here and should be reported to the "Gatsby Starter WordPress" repo for adding better `gatsby-plugin-purgecss` default options.

**Comment by bjfresh** [comment]
Thanks a million! This solved it. Odd that the starter is configured for different output than the production build... seems like a real oversight.

For the record, I solved the problem using the following config:

`
    {
      resolve: 'gatsby-plugin-purgecss', // purges all unused/unreferenced css rules
      options: {
        develop: true,
        printRejected: true,  // Show purgecss rejections in console output
        whitelistPatterns: [/^bm-/, /^swiper-/]  // Whitelist specific selector patterns
      },
    },
`

Big thanks for the assistance.

**Comment by anantoghosh** [comment]
@bjfresh no problem.  
Also make sure to update `gatsby-plugin-purgecss` to v3.1.0 as the `develop` option was introduced in v3.0.0 and `purgeOnly` was included in v3.1.0. [changelog](https://github.com/anantoghosh/gatsby-plugin-purgecss/blob/master/CHANGELOG.md).
Your repo's package.json is still using "gatsby-plugin-purgecss": "^2.4.0".
